### ISSUE TITLE
Terser output code throws "Uncaught reference error", dead code elimination bug

### ISSUE THREAD

**Issue by danieltroger** [terser/terser#1430]
**Bug report or Feature request?** Bug

**Version (complete output of `terser -V` or specific git commit)** terser 5.19.2 & https://try.terser.org/

**Complete CLI command or `minify()` options used**

```
 pbpaste|yarn terser --config-file .terserrc
 ```
 
 Config file:
 ```
 {
  "mangle": {
    "eval": true,
    "properties": {
      "regex": "^(?!__core-js_shared__).*_$"
    }
  },
  "compress": {
    "unsafe": true,
    "arguments": true,
    "ecma": "2020",
    "passes": 4,
    "unsafe_arrows": true
  },
  "format": {
    "ecma": 2020
  }
}

```
 
**`terser` input**

```js
(() => {
  function $67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$consent_from_cookie() {
    return document.cookie.split(";").slice(0);
  }

  const $67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$required_consent_level =
    "statistics";

  // Below variable is unused
  let $67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$has_consent =
    $67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$required_consent_level
      ? /*@__PURE__*/ $67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$consent_from_cookie()?.[
          $67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$required_consent_level
        ] || false
      : true;

  console.log("Hello world");
})();

```

**`terser` output or error**

```
$67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$consent_from_cookie()?.[$67c2532f7fbed187$var$$5dbacdc6a3feeb15$var$required_consent_level],console.log("Hello world");
```

**Expected result**

```
console.log("Hello world")
```

**Comment by fabiosantoscode** [comment]
Thanks for the report! I've crafted a minimal repro. This week I don't have a lot of time to look into this but I'll try

```js
function cookie() {
    return [];
}

var level = "statistics";

var has_consent = level
    ? /*@__PURE__*/ cookie()?.[level]
    : true;

console.log("Hello world");
```

**Comment by fabiosantoscode** [comment]
Released in v5.19.3
